salt/client/__init__.py

```diff
@@ -1698,6 +1699,35 @@ def _prep_pub(self,
 
         return payload_kwargs
 
+    def _prep_jid(self, jid=None, nocache=False, subscribe=False):
+        '''
+        Attempt to generate a jid for this publication if necessary
+        '''
+         # remove empty string
+        jid = jid or None
+
+        # Retrieve the jid
+        fstr = '{0}.prep_jid'.format(self.opts['master_job_cache'])
+        try:
+            # Retrieve the jid
+            jid = self.returners[fstr](nocache=nocache, passed_jid=jid)
+        except (Exception) as exc:
```

Comment:
hmm I'm not sure what to do on this one. since this is passing execution to an arbitrary returner, how do I know what exception class to catch?
I left this as general exception because if it does throw, it will leave jid as '', and send off to master where a second attempt will be made here:
https://github.com/saltstack/salt/blob/develop/salt/master.py#L2209
which seems to only catch KeyError & TypeError. I'm not sure that is desirable either.

Reply:
Ok. Lets leave it then, I also do Not know which in particular to use.